app/models/ability.rb

```diff
@@ -42,6 +42,10 @@ def initialize(user, project: nil, public_token: false)
     end
 
     # Viewing messages with prominence
+    can :read, FoiAttachment do |attachment|
+      can_view_with_prominence?(attachment.prominence, attachment.incoming_message.info_request)
```

Comment:
This doesn't cover the case where the `FoiAttachment` is "normal", but it's `IncomingMessage` is reduced prominence. In this situation you can still view the attachment if you know its direct URL (which search engines will). I think it needs to be like:

```ruby
can_view_with_prominence?(attachment.prominence, attachment.incoming_message.info_request) &&
  can?(:read, attachment.incoming_message)
```

This is pretty critical behaviour, so I think it needs tests, somewhere….